cmd/timestamp_check/main.go

```diff
@@ -0,0 +1,164 @@
+// This tool will initiate an authenticated request generating an initial
+// nonce/timestamp across all supported exchanges and then sleep for 1 minute
+// which will allow us to determine if there are any timestamp issues.
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/signaler"
+)
+
+func main() {
+	flag.BoolVar(&verbose, "verbose", false, "verbose output")
+	flag.Parse()
+
+	log.Println("Checking GCT supported exchanges for timestamp issues on authenticated endpoints.")
+	log.Println()
+
+	if !LoadConfig() {
+		os.Exit(0)
+	}
+
+	var wg sync.WaitGroup
+	for i := range SupportedExchanges {
+		SupportedExchanges[i].SetDefaults()
+		cfg, ok := configs[SupportedExchanges[i].GetName()]
```

Comment:
Instead of duplicating a lot of this logic, you could spawn a basic engine, load the users config (which will have built in auth validation checks, loading of exchanges automatically instead of having to do this) and also have the ability to take advantage of any new exchange additions automatically without having to make hardcoded updates to this tool